lightning/src/ln/peers/handshake/mod.rs

```diff
@@ -0,0 +1,398 @@
+use bitcoin_hashes::{Hash, HashEngine};
+use bitcoin_hashes::sha256::Hash as Sha256;
+use rand::{Rng, thread_rng};
```

Comment:
We don't support rand because it a) often breaks MSRV, and b) doesn't support lots of platforms (read:embedded) that we do, where RNG access is via some proprietary API. Any random values you want need to be passed in from the user.

Reply:
Ok, I'll try to refactor it to have the user pass in entropy.